code/modules/client/client procs.dm

```diff
@@ -310,6 +310,10 @@
 	GLOB.directory[ckey] = src
 
 	//Admin Authorisation
+	// Automatically makes localhost connection an admin
+	var/localhost_addresses = list("127.0.0.1", "::1") // Adresses
+	if(isnull(address) || (address in localhost_addresses))
```

Comment:
There's no way that a player's address can be isnull.

...right?

Reply:
Well it worked when testing so....

Reply:
If there's a way for a player to make their address null, even temporarily, then they can get literally all permissions in-game, and edit their own admin rank to make it permanent across restarts?

So, we'd better hope there isn't a way for this to be null.

Also, I'm curious why the condition isn't "!isnull(address) && address in localhost_addresses.
That prevents a runtime if it IS null, whilst at the same time not handing out admin access like candy if for some reason that does happen.

Yeah, this is pretty paranoid... but given that it gives whoever it affects host-level access, maybe being a bit paranoid is a good idea.

Reply:
I am either gonna make it a config option or do what you did, dont know yet.

Reply:
![](http://www.reactiongifs.com/r/bth.gif)